Removal of allowlist from priviledged tests

horecoli opened the pull request:
## Description
This change updates priviledged test to consider
only containers from CNF under test. As a result of this change, the 'allowlist_helm_chart_container_names' parameter in the cnf-testsuite.yml configuration file is no longer needed.
Therefore this patch also removes this parameter from the configuration, as well as from all samples and examples.

## Issues:
Refs: #1906

## How has this been tested:
 - [ ] Covered by existing integration testing
 - [ ] Added integration testing to cover
 - [ ] Verified all A/C passes
     * [ ] develop
     * [ ] master
     * [ ] tag/other branch
 - [ ] Test environment
    * [ ] Shared Packet K8s cluster
    * [ ] New Packet K8s cluster
    * [ ] Kind cluster
 - [ ] Have not tested

## Types of changes:
 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
 - [x] Documentation update

## Checklist:
**Documentation**
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] No updates required.

**Code Review**
- [ ] Does the test handle fatal exceptions, ie. rescue block

**Issue**
- [ ] Tasks in issue are checked off

HashNuke commented:
Thank you for the updates. Waiting for the build to run for the main branch.

I will re-run this build once the main branch is green - https://github.com/cnti-testcatalog/testsuite/actions/runs/8345433746

taylor commented:
https://github.com/lfn-cnti/bestpractices/blob/main/doc/cbpps/0004-do-not-run-containers-with-privilege-flag.md#notesconstraintscaveats
https://github.com/lfn-cnti/bestpractices/blob/main/doc/cbpps/0003-exceptions.md#summary

taylor commented:
Needs further discussion and updates

taylor commented on CNF_TESTSUITE_YML_USAGE.md:
@horecoli reminder to remove all these changes to this file from the PR.  We will keep the feature to support an allowlist for privileged containers.

horecoli replied:
done

taylor commented on src/tasks/utils/config.cr:
@horecoli these changes should be removed from the PR.  feature is not being removed.

horecoli replied:
done

taylor commented on src/tasks/utils/types/cnf_testsuite_yml_type.cr:
@horecoli remove this update from the PR. keeping feature

horecoli replied:
done

taylor reviewed (changes requested):
1. Remove updates in this PR which remove the feature allowlist -> we are keeping the feature
2. manually test that each example CNFs pass the privilege mode test -> they are not checked by specs

martin-mat commented:
lgtm

horecoli commented:
This change is more or less done, but I am still verifying if all tests are running as expected.

taylor commented:
@horecoli rebase and this should be good to merge.

cc: @agentpoyo @HashNuke

kosstennbl commented:
Some sample CNFs still have empty arrays:
- sample-nonroot
- sample_coredns_default_namespace

Smitholi67 reviewed (approved):
lgtm

horecoli commented:
> Some sample CNFs still have empty arrays:
> 
> * sample-nonroot
> * sample_coredns_default_namespace

Those two samples were added even after I created this PR, so at that time, they were not there. And now I just rebased and forgot to do some grep to check if there is something new. So I fixed it in the latest change. Thank you :)

kosstennbl reviewed (approved):
Needs CI rerun, otherwise - LGTM
